plugins/modules/digital_ocean_kubernetes.py

```diff
@@ -0,0 +1,421 @@
+    type: str
+    aliases: ['API_TOKEN']
+    required: yes
+  state:
+    description:
+      - The usual, C(present) to create, C(absent) to destroy
+    type: str
+    choices: ['present', 'absent']
+    default: present
+    required: yes
+  name:
+    description:
+      - A human-readable name for a Kubernetes cluster.
+    type: str
+    required: yes
+  region:
+    description:
+      - The slug identifier for the region where the Kubernetes cluster will be created.
+    type: str
+    aliases: ['region_id']
+    required: yes
+    default: nyc1
+  version:
+    description:
+      - The slug identifier for the version of Kubernetes used for the cluster. See the /v2/kubernetes/options endpoint for available versions.
+    type: str
+    required: no
+    default: 1.18.8-do.1
+  auto_upgrade:
+    description:
+      - A boolean value indicating whether the cluster will be automatically upgraded to new patch releases during its maintenance window.
+    type: bool
+    required: no
+    default: False
+  surge_upgrade:
+    description:
+      - A boolean value indicating whether surge upgrade is enabled/disabled for the cluster.
+      - Surge upgrade makes cluster upgrades fast and reliable by bringing up new nodes before destroying the outdated nodes.
+    type: bool
+    required: no
+    default: False
+  tags:
+    description:
+      - A flat array of tag names as strings to be applied to the Kubernetes cluster.
+      - All clusters will be automatically tagged "k8s" and "k8s:$K8S_CLUSTER_ID" in addition to any tags provided by the user.
+    type: list
+    required: no
+  maintenance_policy:
+    description:
+      - An object specifying the maintenance window policy for the Kubernetes cluster (see table below).
+    type: dict
+    required: no
+  node_pools:
+    description:
+      - An object specifying the details of the worker nodes available to the Kubernetes cluster (see table below).
+    type: list
+    required: yes
+    default:
+      - name: worker-pool
+        size: s-1vcpu-2gb
+        count: 1
+        tags: []
+        labels: {}
+  vpc_uuid:
+    description:
+      - A string specifying the UUID of the VPC to which the Kubernetes cluster will be assigned.
+      - If excluded, the cluster will be assigned to your account's default VPC for the region.
+    type: str
+    required: no
+  return_kubeconfig:
+    description:
+      - Controls whether or not to return the C(kubeconfig).
+    type: bool
+    required: no
+    default: False
+  wait:
+    description:
+      - Wait for the cluster to be running before returning.
+    type: bool
+    required: no
+    default: True
+  wait_timeout:
+    description:
+      - How long before wait gives up, in seconds, when creating a cluster.
+    type: int
+    default: 600
+requirements:
+    - python >= 2.6
+'''
+
+
+EXAMPLES = r'''
+- name: Create a new DigitalOcean Kubernetes cluster in New York 1
+  community.digitalocean.digital_ocean_kubernetes:
```

Comment:
Just a quick thought: Since this is the first module being added to the collection independent of Ansible core... should we name the module `kubernetes` so it's FQCN is `community.digitalocean.kubernetes`?

Reply:
👍

Reply:
I'll hack this change out soon.

Reply:
https://github.com/ansible-collections/community.digitalocean/pull/25/commits/c9d76ddefed0406ca3eb1ba5c0ddc255ecac7dfd

Reply:
What did happen to this?

Reply:
After not looking at this code for a while, and looking again the other day, I had a change of heart wrt dropping the "redundant digital_ocean prefix", so, I'm making this module match everything else in this collection. Let's give this some thought for 2.0.0, i.e., should things look like `community.digitalocean.digital_ocean_kubernetes` or more simply `community.digitalocean.kubernetes` (applying this same format to every module in here). Personally, I like the latter.